### ISSUE TITLE
Increasing epsilon decreases noise

### ISSUE THREAD

**Issue by echo66** [DataResponsibly/DataSynthesizer#15]
Greetings!

I'm trying to understand your paper and implementation. I've noticed that the more you increase epsilon, the less noise will be generated. In order to understand if that is the expected behavior, I looked into your paper and PrivBayes paper (and, also, a Java implementation) and everyone seems to say that the scale of the noise is given by:

4 * (n_cols - k) / (n_rows * epsilon)

But the definition of differential privacy implies that if epsilon gets closer to 0, there won't be any difference for the query output between the original and synthetic datasets. Am I getting something wrong?

Thanks in advance!

**Comment by haoyueping** [comment]
Thanks for your question! I will update the documentation for epsilon soon.

There are several concepts in Differential Privacy (DP):
1. an original dataset _D1_
2. a neighboring dataset _D2_
3. a randomized algorithm _A_
4. an output _O_
5. a small value _epsilon_

DP requires that 
<a href="https://www.codecogs.com/eqnedit.php?latex=\inline&space;\dpi{200}&space;\fn_cm&space;Pr(A(D1)=O)&space;<&space;e^{epsilon}&space;\cdot&space;Pr(A(D2)=O)" target="_blank"><img src="https://latex.codecogs.com/svg.latex?\inline&space;\dpi{200}&space;\fn_cm&space;Pr(A(D1)=O)&space;<&space;e^{epsilon}&space;\cdot&space;Pr(A(D2)=O)" title="Pr(A(D1)=O) < e^{epsilon} \cdot Pr(A(D2)=O)" /></a>

Under the context of DataSynthesizer, _D1_ is the input dataset, _O_ is the synthetic dataset. 

When _epsilon_=0, Pr(A(D1)=O) = Pr(A(D2)=O). There is no difference between _D1_ and _D2_ for _A_, then _A_ is fully randomized. So reducing _epsilon_ injects more noises.

**Comment by eunbeejang** [comment]
Hello, I'm also trying to understand your work. If you say increased epsilon translates to reducing noise, then what epsilon value will be equivalent to having no DP?

**Comment by haoyueping** [comment]
Hi @eunbeejang ,

The noise required by DP is nearly 0 when the epsilon value is infinity. 

But in the implementation of DataSythesizer, the DP is turned off if epsilon=0

**Comment by amdjedbens** [comment]
> But in the implementation of DataSythesizer, the DP is turned off if epsilon=0.

So with that being said, and in simpler terms, within your practical implementation of DataSynthetizer, you don't strictly adhere to the definition of differential privacy. 
When epsilon is higher, less noise is added, and vice versa. However, when epsilon=0, it means no noise is added at all.
